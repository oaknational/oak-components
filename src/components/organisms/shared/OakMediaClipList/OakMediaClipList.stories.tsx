import React from "react";
import { StoryObj, Meta } from "@storybook/react";

import { OakMediaClipList } from "./OakMediaClipList";

const meta: Meta<typeof OakMediaClipList> = {
  title: "Components/organisms/teacher/OakMediaClipList",
  component: OakMediaClipList,
  tags: ["autodocs"],
  argTypes: {
    lessonTitle: { control: "text" },
  },
  parameters: {
    controls: {
      include: ["lessonTitle"],
    },
  },
};

export default meta;

type Story = StoryObj<typeof OakMediaClipList>;

const mediaClip = {
  clipName: "Standard Clip Name",
  learningCycle: "Learning Cycle",
  timeCode: 657.23,
  imageAltText: "Image Alt Text",
  isAudioClip: false,
  disabled: false,
  thumbnailImage: `https://${process.env.NEXT_PUBLIC_OAK_ASSETS_HOST}/${process.env.NEXT_PUBLIC_OAK_ASSETS_PATH}/v1698336494/samples/landscapes/nature-mountains.jpg`,
  onClick: () => {},
};

export const Default: Story = {
  render: (args) => <OakMediaClipList {...args} />,
  args: {
    lessonTitle: "Lesson title",
    mediaClipList: [
      {
        ...mediaClip,
        muxPlayingState: "played",
      },
      {
        ...mediaClip,
        muxPlayingState: "played",
      },
      {
        ...mediaClip,
        muxPlayingState: "playing",
      },
      {
        ...mediaClip,
        muxPlayingState: "standard",
      },
      {
        ...mediaClip,
        muxPlayingState: "standard",
      },
      {
        ...mediaClip,
        muxPlayingState: "standard",
      },
      {
        ...mediaClip,
        muxPlayingState: "standard",
      },
    ],
  },
};
