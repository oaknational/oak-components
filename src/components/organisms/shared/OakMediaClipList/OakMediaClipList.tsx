import React from "react";

import { OakFlex } from "@/components/atoms/OakFlex";
import { OakTypography, OakUL, OakLI } from "@/components/atoms";
import {
  OakMediaClip,
  OakMediaClipProps,
} from "@/components/organisms/shared/OakMediaClip";
import { OakSolidBorderAccordion } from "@/components/molecules/OakSolidBorderAccordion";

export type OakMediaClipListProps = {
  /**
   * The title of the lesson
   */
  lessonTitle: string;
  /**
   * The array of OakMediaClip props
   */
  mediaClipList: OakMediaClipProps[];
};

type OakMediaClipListHeaderProps = {
  title: string;
  currentClipIndex: number;
  clipCount: number;
};

const MediaClipListHeader = ({
  title,
  currentClipIndex,
  clipCount,
}: OakMediaClipListHeaderProps) => (
  <OakFlex
    $flexDirection={"column"}
    $textAlign={"left"}
    $ph={"inner-padding-m"}
    $pv={"inner-padding-s"}
  >
    <OakTypography $font="body-3" $mb={"space-between-sssx"}>
      Lesson
    </OakTypography>
    <OakTypography $font="heading-7" $mb={"space-between-sssx"}>
      {title}
    </OakTypography>
    <OakTypography $font="body-3">
      {currentClipIndex}/{clipCount} clips
    </OakTypography>
  </OakFlex>
);

/**
 *
 * OakMediaClipList is a scrollable list of OakMediaClip components with clip counter and the title
 *
 */
export const OakMediaClipList = ({
  lessonTitle,
  mediaClipList,
}: OakMediaClipListProps) => {
  const clipCount = mediaClipList.length;
  const currentClipIndex =
    mediaClipList.findIndex(
      (mediaClip) => mediaClip.muxPlayingState === "playing",
    ) + 1;

  const MediaClipListContent = () => (
    <OakUL $reset $ph={"inner-padding-s"} $pt={"inner-padding-xs"}>
      {mediaClipList.map((mediaClipListItem, index) => (
        <OakLI $mb={"space-between-ssx"} key={index}>
          <OakMediaClip {...mediaClipListItem} />
        </OakLI>
      ))}
    </OakUL>
  );

  return (
    <OakSolidBorderAccordion
      header={
        <MediaClipListHeader
          title={lessonTitle}
          currentClipIndex={currentClipIndex}
          clipCount={clipCount}
        />
      }
      children={<MediaClipListContent />}
      id="media-clip-list"
      initialOpen={true}
      $maxHeight={"all-spacing-21"}
    />
  );
};
