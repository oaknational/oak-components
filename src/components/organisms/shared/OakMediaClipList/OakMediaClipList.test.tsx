import React from "react";
import "@testing-library/jest-dom";

import { OakMediaClipList } from "./OakMediaClipList";

import renderWithTheme from "@/test-helpers/renderWithTheme";

describe("OakMediaClipList", () => {
  const defaultProps = {
    lessonTitle: "What is a democratic community?",
    currentClipCounter: 3,
    totalClipCounter: 3,
  };

  it("matches snapshot", () => {
    const { container } = renderWithTheme(
      <OakMediaClipList {...defaultProps} data-testid="test">
        children
      </OakMediaClipList>,
    );
    expect(container).toMatchSnapshot();
  });

  it("renders lesson title correctly", () => {
    const { getByText } = renderWithTheme(
      <OakMediaClipList {...defaultProps}>children</OakMediaClipList>,
    );
    expect(getByText("What is a democratic community?")).toBeInTheDocument();
  });

  it("renders clip counter correctly (both currently playing clip counter and total clip counter)", () => {
    const { getByText } = renderWithTheme(
      <OakMediaClipList {...defaultProps}>children</OakMediaClipList>,
    );
    expect(getByText("3/3 clips")).toBeInTheDocument();
  });
});
