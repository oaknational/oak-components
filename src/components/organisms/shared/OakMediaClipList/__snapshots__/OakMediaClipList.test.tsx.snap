// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OakMediaClipList matches snapshot 1`] = `
[
  .c0 {
  position: relative;
  border-color: #222222;
  border-style: solid;
  font-family: --var(google-font),Lexend,sans-serif;
}

.c3 {
  position: relative;
  max-height: 30rem;
  padding-top: 0rem;
  padding-bottom: 0rem;
  font-family: --var(google-font),Lexend,sans-serif;
}

.c6 {
  font-family: --var(google-font),Lexend,sans-serif;
}

.c6:hover {
  cursor: pointer;
}

.c11 {
  padding-left: 1rem;
  padding-right: 1rem;
  padding-top: 0.75rem;
  padding-bottom: 0.75rem;
  font-family: --var(google-font),Lexend,sans-serif;
  text-align: left;
}

.c12 {
  margin-bottom: 0.25rem;
  font-family: --var(google-font),Lexend,sans-serif;
  font-weight: 300;
  font-size: 0.875rem;
  line-height: 1.25rem;
  -webkit-letter-spacing: -0.005rem;
  -moz-letter-spacing: -0.005rem;
  -ms-letter-spacing: -0.005rem;
  letter-spacing: -0.005rem;
}

.c14 {
  margin-bottom: 0.25rem;
  font-family: --var(google-font),Lexend,sans-serif;
  font-weight: 600;
  font-size: 1rem;
  line-height: 1.25rem;
  -webkit-letter-spacing: 0.0115rem;
  -moz-letter-spacing: 0.0115rem;
  -ms-letter-spacing: 0.0115rem;
  letter-spacing: 0.0115rem;
}

.c16 {
  font-family: --var(google-font),Lexend,sans-serif;
  font-weight: 300;
  font-size: 0.875rem;
  line-height: 1.25rem;
  -webkit-letter-spacing: -0.005rem;
  -moz-letter-spacing: -0.005rem;
  -ms-letter-spacing: -0.005rem;
  letter-spacing: -0.005rem;
}

.c17 {
  position: relative;
  width: 2rem;
  min-width: 2rem;
  height: 2rem;
  min-height: 2rem;
  margin-right: 0.75rem;
  font-family: --var(google-font),Lexend,sans-serif;
}

.c19 {
  overflow: scroll;
  font-family: --var(google-font),Lexend,sans-serif;
  overflow: scroll;
}

.c22 {
  position: relative;
  border-radius: 0.25rem;
  font-family: --var(google-font),Lexend,sans-serif;
}

.c25 {
  position: absolute;
  top: 0rem;
  width: 100%;
  height: 100%;
  border-radius: 0.25rem;
  font-family: --var(google-font),Lexend,sans-serif;
}

.c28 {
  width: 100%;
  font-family: --var(google-font),Lexend,sans-serif;
}

.c30 {
  position: relative;
  width: 6.25rem;
  height: 3.5rem;
  margin-right: 1rem;
  border-radius: 0.125rem;
  font-family: --var(google-font),Lexend,sans-serif;
}

.c33 {
  position: relative;
  width: 100%;
  height: 100%;
  border-radius: 0.125rem;
  font-family: --var(google-font),Lexend,sans-serif;
}

.c34 {
  position: absolute;
  right: 0.25rem;
  bottom: 0.25rem;
  color: #ffffff;
  background: #222222;
  border-radius: 0.125rem;
  font-family: --var(google-font),Lexend,sans-serif;
  font-weight: 300;
  font-size: 0.75rem;
  line-height: 1rem;
  -webkit-letter-spacing: -0.005rem;
  -moz-letter-spacing: -0.005rem;
  -ms-letter-spacing: -0.005rem;
  letter-spacing: -0.005rem;
}

.c42 {
  font-family: --var(google-font),Lexend,sans-serif;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: 0.25rem;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

.c29 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c31 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

.c44 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c18 {
  object-fit: contain;
}

.c38 {
  font-family: --var(google-font),Lexend,sans-serif;
  font-weight: 600;
  font-size: 1rem;
  line-height: 1.25rem;
  -webkit-letter-spacing: 0.0115rem;
  -moz-letter-spacing: 0.0115rem;
  -ms-letter-spacing: 0.0115rem;
  letter-spacing: 0.0115rem;
  text-overflow: ellipsis;
  color: #222222;
}

.c21 {
  margin-bottom: 0.5rem;
  display: revert;
  font-family: --var(google-font),Lexend,sans-serif;
  font-family: --var(google-font),Lexend,sans-serif;
  display: revert;
  display: revert;
}

.c36 {
  font-family: --var(google-font),Lexend,sans-serif;
}

.c39 {
  font-family: --var(google-font),Lexend,sans-serif;
  font-weight: 300;
  font-size: 0.875rem;
  line-height: 1.25rem;
  -webkit-letter-spacing: -0.005rem;
  -moz-letter-spacing: -0.005rem;
  -ms-letter-spacing: -0.005rem;
  letter-spacing: -0.005rem;
  white-space: nowrap;
  text-overflow: ellipsis;
  overflow: hidden;
  color: #575757;
}

.c13 {
  font-family: --var(google-font),Lexend,sans-serif;
  font-weight: 300;
  font-size: 0.875rem;
  line-height: 1.25rem;
  -webkit-letter-spacing: -0.005rem;
  -moz-letter-spacing: -0.005rem;
  -ms-letter-spacing: -0.005rem;
  letter-spacing: -0.005rem;
}

.c15 {
  font-family: --var(google-font),Lexend,sans-serif;
  font-weight: 600;
  font-size: 1rem;
  line-height: 1.25rem;
  -webkit-letter-spacing: 0.0115rem;
  -moz-letter-spacing: 0.0115rem;
  -ms-letter-spacing: 0.0115rem;
  letter-spacing: 0.0115rem;
}

.c20 {
  list-style: none;
  padding: 0;
  margin: 0;
  padding-left: 0.75rem;
  padding-right: 0.75rem;
  padding-top: 0.25rem;
  font-family: --var(google-font),Lexend,sans-serif;
}

.c26 {
  background: none;
  color: inherit;
  border: none;
  padding: 0;
  font: inherit;
  cursor: pointer;
  text-align: left;
  font-family: unset;
  outline: none;
  font-family: --var(google-font),Lexend,sans-serif;
  color: #ffffff;
  background: #ffffff;
  padding: 0.5rem;
  border-color: transparent;
  border-radius: 0.25rem;
}

.c26:disabled {
  pointer-events: none;
  cursor: default;
}

.c40 {
  background: none;
  color: inherit;
  border: none;
  padding: 0;
  font: inherit;
  cursor: pointer;
  text-align: left;
  font-family: unset;
  outline: none;
  font-family: --var(google-font),Lexend,sans-serif;
  color: #222222;
  background: #dff9de;
  padding: 0.5rem;
  border-color: transparent;
  border-radius: 0.25rem;
}

.c40:disabled {
  pointer-events: none;
  cursor: default;
}

.c23 .grey-shadow:has(+ * + .internal-button:focus-visible) {
  box-shadow: 0 0 0 0.3rem rgba(87,87,87,100%);
}

.c23 .yellow-shadow:has(+ .internal-button:focus-visible) {
  box-shadow: 0 0 0 0.125rem rgba(255,229,85,100%);
}

.c23 .grey-shadow:has(+ * + .internal-button:hover) {
  box-shadow: none;
}

.c23 .grey-shadow:has(+ * + .internal-button:active) {
  box-shadow: 0.25rem 0.25rem 0 rgba(87,87,87,100%);
}

.c23 .yellow-shadow:has(+ .internal-button:active) {
  box-shadow: 0.125rem 0.125rem 0 rgba(255,229,85,100%);
}

.c8 {
  font: inherit;
  color: inherit;
  border: none;
  background: none;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  outline: none;
  padding: 0;
}

.c46 {
  position: absolute;
  width: 100%;
  bottom: -0.25rem;
}

.c10 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  width: 100%;
  font-family: --var(google-font),Lexend,sans-serif;
}

.c10:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c5 {
  position: relative;
  max-height: 30rem;
  padding-top: 0rem;
  padding-bottom: 0rem;
  font-family: --var(google-font),Lexend,sans-serif;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: 0.25rem;
}

.c5 .c45 {
  visibility: hidden;
}

.c5 .c9:focus-visible ~ .c45 {
  visibility: visible;
}

.c35 {
  opacity: 0.9;
  padding-left: 2px;
  padding-right: 2px;
}

.c24 {
  min-height: 72px;
  width: 100%;
}

.c32 img {
  border-radius: 2px;
  width: 100%;
  height: 100%;
  object-fit: cover;
}

.c37 {
  position: "relative";
  overflow: hidden;
  text-overflow: ellipsis;
}

.c37 h4 {
  display: -webkit-box;
  -webkit-box-orient: vertical;
  overflow: hidden;
  text-overflow: ellipsis;
  -webkit-line-clamp: 3;
}

.c27 {
  position: relative;
  width: 100%;
  height: 100%;
  display: inline-block;
  -webkit-transition: opacity 300ms ease-out;
  transition: opacity 300ms ease-out;
}

.c27:hover {
  color: #222222;
  background: #f2f2f2;
  border-color: transparent;
}

.c27:hover h4 {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c27:disabled {
  background: #e4e4e4;
  border-color: #808080;
  color: #808080;
}

.c41 {
  position: relative;
  width: 100%;
  height: 100%;
  display: inline-block;
  -webkit-transition: opacity 300ms ease-out;
  transition: opacity 300ms ease-out;
}

.c41:hover {
  color: #222222;
  background: #bef2bd;
  border-color: transparent;
}

.c41:hover h4 {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c41:disabled {
  background: #e4e4e4;
  border-color: #808080;
  color: #808080;
}

.c2 {
  position: relative;
  border-width: 2px;
}

.c43 {
  position: absolute;
  bottom: 0;
  width: 100%;
  height: 50px;
  z-index: 100;
  -webkit-box-shadow: inset 0px -55px 30px -30px rgba(255,255,255,1);
  -moz-box-shadow: inset 0px -55px 30px -30px rgba(255,255,255,1);
  box-shadow: inset 0px -55px 30px -30px rgba(255,255,255,1);
}

<div
    className="c0 c1 c2"
  >
    <div
      className="c3 c4 c5"
    >
      <button
        aria-expanded={true}
        className="c6 c7 c8 c9  c10"
        id="media-clip-list"
        onClick={[Function]}
        type="button"
      >
        <div
          className="c11 c1"
        >
          <div
            className="c12 c13"
          >
            Lesson
          </div>
          <div
            className="c14 c15"
          >
            What is a democratic community?
          </div>
          <div
            className="c16 c13"
          >
            3
            /
            3
             clips
          </div>
        </div>
        <div
          className="c17"
          style={
            {
              "transform": "rotate(180deg)",
              "transition": "all 0.3s ease 0s",
            }
          }
        >
          <img
            alt="An arrow to indicate whether the item is open or closed"
            className="c18"
            data-nimg="fill"
            decoding="async"
            loading="lazy"
            onError={[Function]}
            onLoad={[Function]}
            src="https://res.cloudinary.com/mock-cloudinary-cloud/image/upload/v1699953557/icons/botfld6brychmttwtv6u.svg"
            style={
              {
                "bottom": 0,
                "color": "transparent",
                "height": "100%",
                "left": 0,
                "objectFit": undefined,
                "objectPosition": undefined,
                "position": "absolute",
                "right": 0,
                "top": 0,
                "width": "100%",
              }
            }
          />
        </div>
      </button>
      <div
        aria-labelledby="media-clip-list"
        className="c19"
        hidden={false}
        role="region"
      >
        <ul
          className="c20"
        >
          <li
            className="c21"
          >
            <div
              className="c22 c23 c24"
            >
              <div
                className="c25 grey-shadow"
              />
              <div
                className="c25 yellow-shadow"
              />
              <button
                className="c26 c27 internal-button"
                defaultBorderColor="transparent"
                justifyContent="flex-start"
                muxPlayingState="standard"
                onClick={[Function]}
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
                tabIndex={0}
              >
                <div
                  className="c28 c29"
                >
                  <div
                    className="c30 c31 c32"
                  >
                    <div
                      className="c33"
                      fill={true}
                    >
                      <img
                        alt="Test Image"
                        className="c18"
                        data-nimg="fill"
                        decoding="async"
                        loading="lazy"
                        onError={[Function]}
                        onLoad={[Function]}
                        sizes="100vw"
                        src="/_next/image?url=%2Ftest-image.jpg&w=3840&q=75"
                        srcSet="/_next/image?url=%2Ftest-image.jpg&w=640&q=75 640w, /_next/image?url=%2Ftest-image.jpg&w=750&q=75 750w, /_next/image?url=%2Ftest-image.jpg&w=828&q=75 828w, /_next/image?url=%2Ftest-image.jpg&w=1080&q=75 1080w, /_next/image?url=%2Ftest-image.jpg&w=1200&q=75 1200w, /_next/image?url=%2Ftest-image.jpg&w=1920&q=75 1920w, /_next/image?url=%2Ftest-image.jpg&w=2048&q=75 2048w, /_next/image?url=%2Ftest-image.jpg&w=3840&q=75 3840w"
                        style={
                          {
                            "bottom": 0,
                            "color": "transparent",
                            "height": "100%",
                            "left": 0,
                            "objectFit": undefined,
                            "objectPosition": undefined,
                            "position": "absolute",
                            "right": 0,
                            "top": 0,
                            "width": "100%",
                          }
                        }
                      />
                    </div>
                    <div
                      aria-label="Media clip length: 10:57"
                      className="c34 c35"
                    >
                      <label
                        className="c36"
                      >
                        10:57
                      </label>
                    </div>
                  </div>
                  <div
                    className="c28 c4 c37"
                  >
                    <h4
                      className="c38"
                    >
                      Test Clip
                    </h4>
                    <p
                      className="c39"
                    >
                      Cycle 1
                    </p>
                  </div>
                </div>
              </button>
            </div>
          </li>
          <li
            className="c21"
          >
            <div
              className="c22 c23 c24"
            >
              <div
                className="c25 grey-shadow"
              />
              <div
                className="c25 yellow-shadow"
              />
              <button
                className="c26 c27 internal-button"
                defaultBorderColor="transparent"
                justifyContent="flex-start"
                muxPlayingState="standard"
                onClick={[Function]}
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
                tabIndex={0}
              >
                <div
                  className="c28 c29"
                >
                  <div
                    className="c30 c31 c32"
                  >
                    <div
                      className="c33"
                      fill={true}
                    >
                      <img
                        alt="Test Image"
                        className="c18"
                        data-nimg="fill"
                        decoding="async"
                        loading="lazy"
                        onError={[Function]}
                        onLoad={[Function]}
                        sizes="100vw"
                        src="/_next/image?url=%2Ftest-image.jpg&w=3840&q=75"
                        srcSet="/_next/image?url=%2Ftest-image.jpg&w=640&q=75 640w, /_next/image?url=%2Ftest-image.jpg&w=750&q=75 750w, /_next/image?url=%2Ftest-image.jpg&w=828&q=75 828w, /_next/image?url=%2Ftest-image.jpg&w=1080&q=75 1080w, /_next/image?url=%2Ftest-image.jpg&w=1200&q=75 1200w, /_next/image?url=%2Ftest-image.jpg&w=1920&q=75 1920w, /_next/image?url=%2Ftest-image.jpg&w=2048&q=75 2048w, /_next/image?url=%2Ftest-image.jpg&w=3840&q=75 3840w"
                        style={
                          {
                            "bottom": 0,
                            "color": "transparent",
                            "height": "100%",
                            "left": 0,
                            "objectFit": undefined,
                            "objectPosition": undefined,
                            "position": "absolute",
                            "right": 0,
                            "top": 0,
                            "width": "100%",
                          }
                        }
                      />
                    </div>
                    <div
                      aria-label="Media clip length: 10:57"
                      className="c34 c35"
                    >
                      <label
                        className="c36"
                      >
                        10:57
                      </label>
                    </div>
                  </div>
                  <div
                    className="c28 c4 c37"
                  >
                    <h4
                      className="c38"
                    >
                      Test Clip
                    </h4>
                    <p
                      className="c39"
                    >
                      Cycle 1
                    </p>
                  </div>
                </div>
              </button>
            </div>
          </li>
          <li
            className="c21"
          >
            <div
              className="c22 c23 c24"
            >
              <div
                className="c25 grey-shadow"
              />
              <div
                className="c25 yellow-shadow"
              />
              <button
                className="c40 c41 internal-button"
                defaultBorderColor="transparent"
                justifyContent="flex-start"
                muxPlayingState="playing"
                onClick={[Function]}
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
                tabIndex={0}
              >
                <div
                  className="c28 c29"
                >
                  <div
                    className="c30 c31 c32"
                  >
                    <div
                      className="c33"
                      fill={true}
                    >
                      <img
                        alt="Test Image"
                        className="c18"
                        data-nimg="fill"
                        decoding="async"
                        loading="lazy"
                        onError={[Function]}
                        onLoad={[Function]}
                        sizes="100vw"
                        src="/_next/image?url=%2Ftest-image.jpg&w=3840&q=75"
                        srcSet="/_next/image?url=%2Ftest-image.jpg&w=640&q=75 640w, /_next/image?url=%2Ftest-image.jpg&w=750&q=75 750w, /_next/image?url=%2Ftest-image.jpg&w=828&q=75 828w, /_next/image?url=%2Ftest-image.jpg&w=1080&q=75 1080w, /_next/image?url=%2Ftest-image.jpg&w=1200&q=75 1200w, /_next/image?url=%2Ftest-image.jpg&w=1920&q=75 1920w, /_next/image?url=%2Ftest-image.jpg&w=2048&q=75 2048w, /_next/image?url=%2Ftest-image.jpg&w=3840&q=75 3840w"
                        style={
                          {
                            "bottom": 0,
                            "color": "transparent",
                            "height": "100%",
                            "left": 0,
                            "objectFit": undefined,
                            "objectPosition": undefined,
                            "position": "absolute",
                            "right": 0,
                            "top": 0,
                            "width": "100%",
                          }
                        }
                      />
                    </div>
                    <div
                      aria-label="Media clip length: 10:57"
                      className="c34 c35"
                    >
                      <label
                        className="c36"
                      >
                        10:57
                      </label>
                    </div>
                  </div>
                  <div
                    className="c28 c4 c37"
                  >
                    <h4
                      className="c38"
                    >
                      Test Clip
                    </h4>
                    <p
                      className="c39"
                    >
                      Cycle 1
                    </p>
                  </div>
                </div>
              </button>
            </div>
          </li>
        </ul>
        <div
          className="c42 c43"
        />
      </div>
      <div
        className="c42 c44 c45 c46"
      >
        <svg
          className=""
          height="7"
          preserveAspectRatio="none"
          viewBox="0 0 195 7"
          width="100%"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M193.989 5.27816C191.224 5.83961 188.29 5.95094 185.233 5.95436C164.998 5.97909 144.806 6.04167 124.593 6.02065C103.549 6.00315 82.5133 5.8886 61.4695 5.85778C45.3904 5.83449 29.2961 5.89496 13.2098 5.8717C10.2284 5.84036 7.27134 5.73181 4.38119 5.54762C1.78065 5.40454 0.188712 4.70174 0.482863 4.00998C0.590325 3.75687 0.613986 3.50218 0.553519 3.24879C0.431797 3.00904 0.257402 2.77161 0.0312099 2.53766C1.56714 2.43619 3.12843 2.36598 4.70115 2.32768C16.4883 2.26831 28.2976 2.13275 40.0693 2.2123C52.4335 2.29696 64.7916 2.21996 77.1492 2.22285C105.42 2.23007 133.688 2.24999 161.954 2.2826C168.927 2.29305 175.886 2.42911 182.844 2.53093C184.644 2.56126 186.424 2.63781 188.161 2.75952C191.997 3.02103 193.82 3.68671 193.928 4.7401C193.905 4.90949 193.955 5.06526 193.989 5.27816ZM7.54114 5.28886L7.54209 5.15951L5.46075 5.16831C5.46037 5.21968 5.46 5.27104 5.4596 5.3243L7.54114 5.28886ZM7.31844 3.14218C6.36666 3.83292 6.36668 3.83291 7.94956 3.79958C7.77774 3.57965 7.56255 3.35991 7.35458 3.14203L7.31844 3.14218ZM58.0677 2.64406L58.2202 2.55021L54.6067 2.56549L54.606 2.6587L58.0677 2.64406ZM59.2655 3.87356L59.3735 3.93587L61.2886 3.92776L61.2891 3.865L59.2655 3.87356ZM4.15738 2.56776L4.03376 2.6729L6.74385 2.66144L6.74462 2.55681L4.15738 2.56776Z"
            fill="#222222"
          />
          <path
            d="M194.048 3.7872C191.283 4.34864 188.349 4.45997 185.292 4.46339C165.059 4.48811 144.869 4.55069 124.658 4.52966C103.616 4.51215 82.5815 4.39759 61.5395 4.36676C45.4617 4.34347 29.3688 4.40393 13.2838 4.38067C10.3027 4.34933 7.34591 4.24078 4.45601 4.05658C1.85568 3.9135 0.263876 3.2107 0.558012 2.51894C0.665459 2.26584 0.68912 2.01114 0.628653 1.75775C0.506946 1.51801 0.332582 1.28057 0.10639 1.04662C1.6422 0.945149 3.20335 0.874948 4.77593 0.836643C16.562 0.777274 28.3704 0.641721 40.141 0.721281C52.5042 0.805943 64.8613 0.728942 77.2178 0.731837C105.486 0.739072 133.752 0.759003 162.015 0.791623C168.988 0.802073 175.946 0.938139 182.904 1.03996C184.703 1.07029 186.484 1.14684 188.22 1.26856C192.055 1.53006 193.879 2.19575 193.987 3.24913C193.964 3.41853 194.013 3.57429 194.048 3.7872ZM7.6157 3.79783L7.61665 3.66848L5.53547 3.67728C5.5351 3.72864 5.53472 3.78001 5.53433 3.83327L7.6157 3.79783ZM7.39301 1.65115C6.44131 2.34188 6.44133 2.34188 8.02407 2.30855C7.85227 2.08862 7.63709 1.86887 7.42914 1.651L7.39301 1.65115ZM58.1379 1.15304L58.2904 1.05919L54.6772 1.07448L54.6765 1.16769L58.1379 1.15304ZM59.3357 2.38254L59.4436 2.44485L61.3585 2.43675L61.359 2.37398L59.3357 2.38254ZM4.23221 1.07672L4.10861 1.18187L6.81847 1.1704L6.81924 1.06578L4.23221 1.07672Z"
            fill="#FBD60E"
          />
        </svg>
      </div>
    </div>
  </div>,
  ",",
]
`;
